jberet-core/src/main/java/org/jberet/operations/ForceStopJobOperatorImpl.java

```diff
@@ -0,0 +1,13 @@
+package org.jberet.operations;
+
+import jakarta.batch.runtime.BatchStatus;
+import org.jberet.runtime.JobExecutionImpl;
+
+public class ForceStopJobOperatorImpl extends DefaultJobOperatorImpl {
+    public void forceStop(final long executionId) {
+        final JobExecutionImpl jobExecution = getJobExecutionImpl(executionId);
+        jobExecution.setBatchStatus(BatchStatus.STOPPED);
```

Comment:
Maybe we can set the `exitstatus` into `CRASHED`.

And we need to update the `endtime` to the current time.

We also need to update the other tables here:

- `partition_execution`
- `step_execution`

Reply:
This isn't thread safe. You'd really need all this to happen in something similar to a transaction which either all or none are updated.

Reply:
Yes, I should put them into a single tx like the implementation in the current `stop()` method in `JdbcRepository` :

```java
   @Override
    public void stopJobExecution(final JobExecutionImpl jobExecution) {
        super.stopJobExecution(jobExecution);
        final String[] stopExecutionSqls = {
                sqls.getProperty(STOP_JOB_EXECUTION),
                sqls.getProperty(STOP_STEP_EXECUTION),
                sqls.getProperty(STOP_PARTITION_EXECUTION)
        };
        final String jobExecutionIdString = String.valueOf(jobExecution.getExecutionId());
        final String newBatchStatus = BatchStatus.STOPPING.toString();
        final Connection connection = getConnection();
        Statement stmt = null;
        try {
            stmt = connection.createStatement();
            for (String sql : stopExecutionSqls) {
                stmt.addBatch(sql.replace("?", jobExecutionIdString));
            }
            stmt.executeBatch();
        } catch (Exception e) {
            throw BatchMessages.MESSAGES.failToRunQuery(e, Arrays.toString(stopExecutionSqls));
        } finally {
            close(connection, stmt, null, null);
        }
    }
```